### PR TITLE
feat: docs, mdx, layout package

### PR DESCRIPTION
**Pull request by koca**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been reviewed and added / updated if needed (for bug fixes
      /start features)

## Pull request type

Please check the type of change your PR introduces:

- [X] Bugfix
- [X] Feature
- [X] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

#### Updates 

- add mdx support to docs using vite-plugin-mdx-vue - `website/`
- add `useColorModeValue` api - `packages/c-color-mode/color-mode-provider.ts`
- add `@chakra-ui/layout` package - `packages/layout`
- add support for global components like `router-link` and add tests for 3 cases (dom elements, vue components, global components) - `packages/system/chakra.ts` 
- update deps
- add `getValidChildren` helper
- add reusuable `themingProps`
- add core types

#### New Components

`c-aspect-ratio`
`c-badge`
`c-box`
`c-center`
`c-container`
`c-divider`
`c-grid`
`c-heading`
`c-kbd`
`c-link-box`
`c-link`
`c-list`
`c-simple-grid`
`c-spacer`
`c-stack`
`c-text`
`c-wrap`

## Does this introduce a breaking change?

- [X] Yes
- [ ] No

Breaking Changes:

moved c-badge to layout 
moved c-code to layout 
moved c-flex to layout

### CODE REVIEW

File: website/src/docs-theme/components/AppHeader.vue [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+</template>
+
+<script setup lang="ts">
+import { useColorModeValue } from '@chakra-ui/c-color-mode'
+import { useWindowScroll } from '@vueuse/core'
+import { onMounted, ref, watchEffect } from 'vue'
+
+const headerRef = ref<{ $el: HTMLDivElement }>()
+
+const bg = useColorModeValue('white', 'gray.800')
```
Comment (carwack) [inline website/src/docs-theme/components/AppHeader.vue:82]: This is sweet! 
I was playing with colorMode myself and ended up with ```:bg="`${colorMode}.bg`"``` but this looks interesting to use.

File: website/src/docs-theme/components/AppHeader.vue [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+const height = ref(0)
+onMounted(() => {
+  height.value = headerRef.value?.$el.getBoundingClientRect().height ?? 0
+})
+
+watchEffect(() => {
+  console.log({ y: y.value })
+  console.log({ height: height.value })
+})
```
Comment (carwack) [inline website/src/docs-theme/components/AppHeader.vue:93]: Is this still needed? 😊

File: website/src/docs-theme/components/AppHeader.vue [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          justify="flex-end"
+          w="100%"
+          maxW="824px"
+          align="center"
+          color="gray.400"
+        >
+          <chakra.div>Search</chakra.div>
+          <chakra.div>Switcher</chakra.div>
+          <CFlex
+            todo="HStack"
```
Comment (carwack) [inline website/src/docs-theme/components/AppHeader.vue:40]: what does the ```todo``` prop do? 😊

File: website/src/docs-theme/components/Logo.vue [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      </linearGradient>
+    </defs>
+  </chakra.svg>
+</template>
+
+<script setup lang="ts">
+import { useColorModeValue } from '@chakra-ui/c-color-mode'
+
+const fill = useColorModeValue('#2D3748', '#fff')
+console.log({ fill })
```
Comment (carwack) [inline website/src/docs-theme/components/Logo.vue:41]: Looks like this console.log isn't needed anymore right?

File: website/src/docs-theme/components/AppHeader.vue (hunk 1 above)
Comment (koca) [reply website/src/docs-theme/components/AppHeader.vue:82]: yep this api is from chakraui https://github.com/chakra-ui/chakra-ui/blob/main/packages/color-mode/src/color-mode-provider.tsx#L187 ^^

File: website/src/docs-theme/components/AppHeader.vue (hunk 2 above)
Comment (koca) [reply website/src/docs-theme/components/AppHeader.vue:93]: don't worry about the console.logs until the PR is ready for merge :) this PR is a draft and i'll remove all console.log or todos etc in the end :)

File: website/src/docs-theme/components/AppHeader.vue (hunk 3 above)
Comment (koca) [reply website/src/docs-theme/components/AppHeader.vue:40]: again just ignore it, it was a todo for myself :D

File: website/src/docs-theme/components/AppHeader.vue (hunk 2 above)
Comment (koca) [reply website/src/docs-theme/components/AppHeader.vue:93]: i don't think this is a reactive though i'll probably change the implementation.

File: website/src/docs-theme/components/AppHeader.vue (hunk 1 above)
Comment (koca) [reply website/src/docs-theme/components/AppHeader.vue:82]: i don't think this is a reactive though i'll probably change the implementation.

File: packages/layout/package.json [hunk 5]
```diff
@@ -37,14 +40,12 @@
...
   "dependencies": {
-    "@chakra-ui/utils": "^1.5.0",
-    "@chakra-ui/vue-system": "0.1.0-alpha.1"
-  },
-  "devDependencies": {
-    "vue": ">=3.0.5"
+    "@chakra-ui/styled-system": "^1.9.0",
+    "@chakra-ui/vue-system": "0.1.0-alpha.1",
+    "@chakra-ui/vue-utils": "0.1.0-alpha.1",
+    "@chakra-ui/vue-layout": "0.1.0-alpha.1"
```
Comment (codebender828) [inline packages/layout/package.json:46]: This might be a circular dependency. The `@chakra-ui/vue-layout` package is requiring itself. 🤔

Comment (koca) [reply packages/layout/package.json:46]: it was late man :D fixed

File: packages/layout/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { ThemingProps } from '@chakra-ui/vue-system'
+import { PropType } from 'vue'
+
+export const vueThemingProps = {
+  colorScheme: String as PropType<ThemingProps['colorScheme']>,
+  variant: String as PropType<ThemingProps['variant']>,
+  size: String as PropType<ThemingProps['size']>,
+  styleConfig: String as PropType<ThemingProps['styleConfig']>,
+}
```
[the remark this answers is not in the document]
Comment (koca) [reply packages/layout/src/utils.ts:9]: yep agree moved to `vue-utils` but `themingProps` is already used in components we need to think something else

**Comment by koca** [comment]
i moved `c-badge` to layout folder but `c-flex` and `c-code` needs to be moved. But we can do it later.

**Review by codebender828 — APPROVED** [review]
Solid PR, @koca 🎉💖

Thanks for the organizational things you included in here too. Will add the changeset and merge thereafter

**Pull request merged**